examples/datepicker/datepicker.dateValue.tsx

```diff
@@ -0,0 +1,54 @@
+import { type CodeData } from '~/components/code-demo';
+import { Datepicker } from '~/src';
+
+const code = `
+'use client';
+
+import { Datepicker } from 'flowbite-react';
+
+function Component() {
+  return (
+    <Datepicker
+      dateValue={new Date(2022, 11, 25)} // 25th December 2022
+    />
+  );
+}
+`;
+
+const codeRSC = `
+import { Datepicker } from 'flowbite-react';
+
+function Component() {
+  return (
+    <Datepicker
+      dateValue={new Date(2022, 11, 25);} // 25th December 2022
+    />
+  );
+}
+`;
+
+function Component() {
+  return (
+    <Datepicker
+      dateValue={new Date(2022, 11, 25)} // 25th December 2022
```

Comment:
Thanks for creating this! Why don't we use the name [`defaultValue`](https://react.dev/reference/react-dom/components/select#providing-an-initially-selected-option)?

Reply:
Thank you for the review, happy new years. 

- defaultDate is the name for the date loaded by default.
- dateValue is the name for the modified date-pickers date value.

In naming terms, defaultValue doesn't seem to catch the idea of changing as dateValue would do.

IMO, would eliminate defaultDate and use only dateValue (because it is just an update for defaultDate), **BUT** that would affect prior users and that would be problematic so I kept it as a new prop.

And the opposite ( which I guess was what you thought ), eliminating dateValue and using defaultDate, would work, but wouldn't have a friendly naming IMO.

So, for now, my idea on this PR was to keep both properties, and further on make a transition from one to another if necessary, or deprecate defaultValue with deprecation warnings during some time, or something like that...

Reply:
Would that make sense @tulup-conner ?

Reply:
IMO the best DX here is to convert `defaultDate` to `defaultValue` and `dateValue` to `value`. These are universal React APIs for form elements. People expect them to work intuitively inside forms. It's implied when you create a `<Datepicker defaultValue={}>` that the default value is a date, because it's a datepicker. Etc.

Reply:
Agreed with @tulup-conner on this.

`defaultValue` and `value` are the defacto standard.

Reply:
Okay, that's good @SutuSebastian, @tulup-conner. Appreciate it.
Then we'll have a breaking change here in this case. 

Shall I create a deprecation warning message inside the component?  I was thinking of something like:

```
 useEffect(() => {
    if (defaultDate !== undefined) {
      logDeprecationWarning('MyComponent', 'deprecatedProp', 'newProp');
    }
  }, [deprecatedProp]);
```
Would that work? Any suggestions/procedures on breaking changes I should head up?

Reply:
We'll provide a clear message in the breaking changes section of the release it is in. This project is under development actively and we disclose that clearly so people know to expect breaking changes unpredictably :)

Reply:
Dun @tulup-conner.